Nodes/SerialIO.c

```diff
@@ -306,8 +309,14 @@ char gettekn(void) {
       UpdateInactive();
     }
     if((signals & serreadsig) && CheckIO((struct IORequest *)serreadreq)) {
-      ch = sergettkn();
-      ConvChrsToAmiga(&ch,1,Servermem->inne[nodnr].chrset);
+      int conv;
+      serbuf[seridx] = sergettkn();
+      conv = ConvMBChrsToAmiga(&ch,serbuf,seridx+1,
+                               Servermem->inne[nodnr].chrset);
+      if (conv < 0) {
+        ch = 0;
```

Comment:
Not strictly needed with the current code, but I thought it would be more readable to explicitly clear it, instead of implicitly relying on ConvMBChrsToAmiga not modifying it when returning negative. I can remove it if you still want?